content/blog/2019/11/2019-11-22-welcome-to-the-matrix.adoc

```diff
@@ -0,0 +1,394 @@
+                        axis {
+                            name 'BROWSER'
+                            values 'edge'
+                        }
+                    }
+                }
+                stages {
+                    stage('Build') {
+                        steps {
+                            echo "Do Build for ${PLATFORM} - ${BROWSER}"
+                        }
+                    }
+                    stage('Test') {
+                        steps {
+                            echo "Do Test for ${PLATFORM} - ${BROWSER}"
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+----
+
+
+If I run this Pipeline from the Jenkins UI and set the `PLATFORM_FILTER` parameter to `mac`, I'll get something like the output below:
+
+.Log output (truncated - PLATFORM_FILTER = 'mac' )
+[source]
+----
+...
+[Pipeline] stage
+[Pipeline] { (BuildAndTest)
+[Pipeline] parallel
+[Pipeline] { (Branch: Matrix - OS = 'linux', BROWSER = 'firefox')
+[Pipeline] { (Branch: Matrix - OS = 'windows', BROWSER = 'firefox')
+[Pipeline] { (Branch: Matrix - OS = 'mac', BROWSER = 'firefox')
+[Pipeline] { (Branch: Matrix - OS = 'linux', BROWSER = 'chrome')
+[Pipeline] { (Branch: Matrix - OS = 'windows', BROWSER = 'chrome')
+[Pipeline] { (Branch: Matrix - OS = 'mac', BROWSER = 'chrome')
+[Pipeline] { (Branch: Matrix - OS = 'windows', BROWSER = 'safari')
+[Pipeline] { (Branch: Matrix - OS = 'mac', BROWSER = 'safari')
+[Pipeline] { (Branch: Matrix - OS = 'windows', BROWSER = 'edge')
+...
+Stage "Matrix - OS = 'linux', BROWSER = 'chrome'" skipped due to when conditional
+Stage "Matrix - OS = 'linux', BROWSER = 'firefox'" skipped due to when conditional
+...
+Do Build for mac - firefox
+Do Build for mac - chrome
+Do Build for mac - safari
+...
+Stage "Matrix - OS = 'windows', BROWSER = 'chrome'" skipped due to when conditional
+Stage "Matrix - OS = 'windows', BROWSER = 'edge'" skipped due to when conditional
+...
+Do Test for mac - safari
+Do Test for mac - firefox
+Do Test for mac - chrome
+----
+
+IMPORTANT: Come join me at link:https://www.cloudbees.com/devops-world/lisbon[DevOps World | Jenkins World 2019] for "link:https://sched.co/UeQe[**Declarative Pipeline 2019: Tips, Tricks and What's Next**]".  
+I'll go over what's been added to Pipeline in the last year (including matrix) and discuss ideas about where pipeline should go next.
+
+== Conclusion 
+
+In this blog post, we've looked at how to use the `matrix` directive to make concise but powerful declarative pipelines.  
+An equivalent pipeline created without `matrix` would easily be several times larger, and much harder to understand and maintain. 
+
+Matrix is now available from the experimental update center. 
+It will be released to the main update center as soon as we're done putting the finishing touches on the documentation and online help. 
+
+== Links
+
+* link:https://updates.jenkins.io/experimental/[Jenkins Experimental Update Center]
```

Comment:
Also link https://jenkins.io/doc/developer/publishing/releasing-experimental-updates/#using-the-experimental-update-center ?

Reply:
```suggestion
* link:https://updates.jenkins.io/experimental/[Jenkins Experimental Update Center]
* link:https://jenkins.io/doc/developer/publishing/releasing-experimental-updates/#using-the-experimental-update-center[Using the Jenkins Experimental Update Center]
 
```